runtime/lua/vim/fs.lua

```diff
@@ -356,8 +370,10 @@ function M.find(names, opts)
         end
 
         if
-          type_ == 'directory'
-          or (type_ == 'link' and opts.follow and (vim.uv.fs_stat(f) or {}).type == 'directory')
+          (
+            type_ == 'directory'
+            or (type_ == 'link' and opts.follow and (vim.uv.fs_stat(f) or {}).type == 'directory')
+          ) and (not opts.filter or opts.filter(f) ~= false)
```

Comment:
So if filter returns `nil` that's considered truthy? Is that how `vim.fs.dir` works?

Reply:
This is the case now to keep the the same behavior of `skip` just with the new parameter, I can change filter to use falsey values instead but that would mean having a slightly more complex check in `vim.fs.dir`, now both functions would have different behaviors. 

How is it usually done in that case ? should the function error out if both `filter` and `skip` are provided or should it just perform both checks giving preference to `filter` ?

Reply:
> now both functions would have different behaviors.

whatever matches the current behavior of vim.fs.dir is preferred

Reply:
ok then, let's keep the same behavior as `skip` at least for now,  let me know if more changes are require to move the PR forward